contracts/cw-law-stone/src/msg.rs

```diff
@@ -0,0 +1,52 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::Binary;
+#[allow(unused_imports)]
+use logic_bindings::AskResponse;
+
+/// Instantiate message
+#[cw_serde]
+pub struct InstantiateMsg {
+    /// The Prolog program carrying law rules and facts.
+    pub program: Binary,
+
+    /// The `cw-storage` contract address on which to store the law program.
+    pub storage_address: String,
+}
+
+/// Execute messages
+#[cw_serde]
+pub enum ExecuteMsg {
+    /// # BreakStone
+    /// Break the stone making this contract unusable, by clearing all the related resources:
+    /// - Unpin all the pinned objects on `cw-storage` contracts, if any.
+    /// - Forget the main program (i.e. or at least unpin it).
+    /// Only the contract admin is authorized to break it, if any.
+    /// If already broken, this is a no-op.
+    BreakStone,
+}
+
+/// Query messages
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    /// # Ask
+    /// If not broken, ask the logic module the provided query with the law program loaded.
+    #[returns(AskResponse)]
+    Ask { query: String },
+
+    /// # Program
+    /// If not broken, returns the law program location information.
+    #[returns(ProgramResponse)]
+    Program,
```

Comment:
@amimart, @ccamel Just a question, Wouldn't it be nice to have a query that returns the list of linked resources ? Maybe useless ?

Reply:
Yes this information is already present in law program but at the first-level of depth, if a program call another program that call another program... This isn't visible at this level of depth. But since this dependencies is kept into a field in state, this can be easily added later, the time to see if it's necessary :p

Reply:
Yes, I completely agree with you guys. At first glance, it seems like an interesting idea, but let's take the time to evaluate its relevance. @bdeneux can you open an issue for this one? Thanks :pray: